File: python solution code/check_two_string_is_anargram_or_not_using _bit_manipulation.py

def checkAnargram(s1, s2):
    l1, l2 = len(s1), len(s2)

    if l1 != l2:
        return False
    val = 0
    for i in range(0,l1):
        val ^=  ord(s1[i])
        val ^=  ord(s2[i])
    return val == 0


s1 = input()
s2 = input()
if(checkAnargram(s1,s2)):
    print("Yes")
else:
    print("No")

File: C++ solution code/check_two_string_is_anargram_or_not_using_bit_manipulation.cpp
#include<bits/stdc++.h>
using namespace std;

bool checkAnargram(string s1, string s2)
{
    int l1 = s1.size(), l2 = s2.size();

    if(l1 != l2) return false;

    int val = 0;
    for(int i = 0; i < l1; i++) {
        val ^= (int)s1[i];
        val ^= (int)s2[i];
    }
    return val == 0;
}

int main()
{
    string s1, s2;
    cin >> s1 >> s2;
    cout << (checkAnargram(s1, s2)?"YES":"NO") << endl;
    return 0;
}